hr_holidays_leave_auto_approve/models/hr_holidays.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# © 2016 ONESTEiN BV (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, values):
+        result = super(HrHolidays, self).create(values)
+        if result.holiday_status_id.auto_approve:
+            result.sudo().signal_workflow('validate')
```

Comment:
Even if the transition was automatically triggered from `create`, this would always call a `write` passing as only value the `state`. That's because the function `holidays_validate` (called by the workflow transition) is defined as following:

```
    def holidays_validate(self, cr, uid, ids, context=None):
        obj_emp = self.pool.get('hr.employee')
        ids2 = obj_emp.search(cr, uid, [('user_id', '=', uid)])
        manager = ids2 and ids2[0] or False
        **self.write(cr, uid, ids, {'state': 'validate'}, context=context)**
        ...
```
So, unless we don't define a brand new method to be invoked from the new workflow transition in order to perform the state change, this will always be an issue. In my opinion it's clear that defining and using a method different from `holidays_validate` is not a good idea, as it would fork the standard workflow and break the inheritance chain.